sql/core/src/test/scala/org/apache/spark/sql/DataFrameStatSuite.scala

```diff
@@ -366,6 +366,27 @@ class DataFrameStatSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  test("SPARK-28818: Respect original column nullability in `freqItems`") {
+    val rows = spark.sparkContext.parallelize(
+      Seq(Row("1", "a"), Row("2", null), Row("3", "b"))
+    )
+    val schema = StructType(Seq(
+      StructField("non_null", StringType, false),
+      StructField("nullable", StringType, true)
+    ))
+    val df = spark.createDataFrame(rows, schema)
+
+    val result = df.stat.freqItems(df.columns)
+
+    val nonNullableFreqItems = result.schema("non_null_freqItems").dataType.asInstanceOf[ArrayType]
+    val nullableFreqItems = result.schema("nullable_freqItems").dataType.asInstanceOf[ArrayType]
+
+    assert(nonNullableFreqItems.containsNull == false)
+    assert(nullableFreqItems.containsNull == true)
+    // Original bug was a NullPointerException exception caused by calling collect(), test for this
+    val resultRows = result.collect()
```

Comment:
Also, please fix `val resultRows = result.collect()` -> `result.collect()` if you plan to test nullability only in this test case.